oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/changesetenricher/ChangesetEnricher.java

```diff
@@ -0,0 +1,164 @@
+package org.heigit.bigspatialdata.oshdb.util.changesetenricher;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.heigit.bigspatialdata.oshdb.util.OSHDBBoundingBox;
+import org.heigit.bigspatialdata.oshdb.util.OSHDBTimestamp;
+import org.slf4j.LoggerFactory;
+
+public class ChangesetEnricher {
+  private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(ChangesetEnricher.class);
+
+  private final Connection conn;
+  private final Map<Long, OSMChangeset> changests;
+
+  /**
+   * This class works as a cache (like
+   * {@link org.heigit.bigspatialdata.oshdb.util.tagtranslator.TagTranslator}). Attention: The
+   * ChangesetDatabase is minutely updated but the cache isn't. So subsequent calls to this Class
+   * for the same ID will always return the same result. If you want to update the cache you will
+   * have to create a new instance.
+   *
+   * @param conn The connection to the ohsome-changeset postgres-database.
+   */
+  public ChangesetEnricher(Connection conn) {
+    this.changests = new ConcurrentHashMap<>(0);
+    this.conn = conn;
+  }
+
+  /**
+   * Get latest update Timestamp of Changeset-Database (may be null if unknown).
+   *
+   * @return
+   * @throws SQLException
+   */
+  public OSHDBTimestamp getValidUntil() throws SQLException {
+
+    Timestamp timestamp;
+    try (PreparedStatement prepareStatement = this.conn.prepareStatement("SELECT last_timestamp FROM osm_changeset_state")) {
+      ResultSet executeQuery = prepareStatement.executeQuery();
+      executeQuery.next();
+      timestamp = executeQuery.getTimestamp(1);
+    }
+    if (timestamp == null) {
+      return null;
+    }
+    return new OSHDBTimestamp(timestamp);
+  }
+
+  /**
+   * Get a Changeset with comments. May be null if changesetId not in Database. Any field may also
+   * be null.
+   *
+   * @param changesetId
+   * @return
+   */
+  public OSMChangeset getChangeset(long changesetId) throws SQLException {
+    if (this.changests.containsKey(changesetId)) {
+      return this.changests.get(changesetId);
+    }
+    OSMChangeset changeset;
+    try (PreparedStatement prepareStatement = this.conn.prepareStatement(
+        "SELECT user_id,created_at,min_lat,max_lat,min_lon,max_lon,closed_at,open,num_changes,user_name,tags "
+        + "FROM osm_changeset "
+        + "WHERE id = ? ;")) {
+      prepareStatement.setLong(1, changesetId);
+      ResultSet resultSet = prepareStatement.executeQuery();
+
+      if (resultSet.next()) {
+        Long user_id = resultSet.getLong("user_id");
+
+        OSHDBTimestamp created_at;
+        try {
+          created_at = new OSHDBTimestamp(resultSet.getTimestamp("created_at"));
```

Comment:
don't use NullPointerException to check for null!
Please test explicit for it like 
```java
Timestamp ts = resultSet.getTimestamp("created_at");
if(ts != null) {
  created_at = new OSHDBTimestamp(ts);
}else {
 created_at = null;
}
```